pkg/rhos/ocpgwdeployer.go

```diff
@@ -316,11 +316,7 @@ func (d *ocpGatewayDeployer) Cleanup(status reporter.Interface) error {
 	gwNodes := gwNodesList.Items
 
 	for i := range gwNodes {
-		// Check if the instance belongs to the cluster (identified via infraID) we are operating on.
-		// If the instance name matches with d.InfraID + "-submariner-gw-", it implies that
-		// the gateway node was deployed using the OCPMachineSet which is already handled above and
-		// the node should be already be in deleting state.
-		if !strings.HasPrefix(gwNodes[i].Name, d.InfraID) || strings.HasPrefix(gwNodes[i].Name, prefix) {
+		if gwNodes[i].GetLabels()[ocp.MachineSetOwnerLabel] == "submariner" {
```

Comment:
Should make "submariner" a constant in `ocp` as it's referenced in several places, eg `LabelValueOwnedBy`.